### PR TITLE
DRAFT:  add placeholder id to placeholder atom

### PR DESCRIPTION
**Pull request by caleb-flores**

### CODE REVIEW

File: src/editor-mathfield/keyboard-input.ts [hunk 1]
```diff
@@ -363,10 +362,8 @@ export function onTypedText(
 ): void {
   const { model } = mathfield;
   if (mathfield.options.readOnly) {
-    model.announce('plonk');
```
Comment (arnog) [inline src/editor-mathfield/keyboard-input.ts:366]: Why did you remove this?

File: src/editor-mathfield/mode-editor-math.ts [hunk 2]
```diff
@@ -193,6 +196,38 @@ export class MathModeEditor extends ModeEditor {
...
+    newPlaceholders.forEach((placeholder) => {
+      if (
+        placeholder.placeholderId &&
+        !model.mathfield._placeholders.has(placeholder.placeholderId)
+      ) {
+        const element = new MathfieldElement();
+        element.classList.add('mathfield');
+        element.style.display = 'inline-block';
+        element.style.zIndex = '1001';
+        element.style.fontSize = '60%';
```
Comment (arnog) [inline src/editor-mathfield/mode-editor-math.ts]: Why do you need to change the fontSize?

File: src/editor-mathfield/pointer-input.ts [hunk 3]
```diff
@@ -19,7 +19,12 @@ export function onPointerDown(
 ): void {
   //Reset the atom bounds cache
   mathfield._atomBoundsCache = new Map<string, Rect>();
-
+  if (
+    (evt.target as HTMLElement).tagName === 'MATH-FIELD' &&
+    mathfield.options.readOnly
+  ) {
+    return;
```
Comment (arnog) [inline src/editor-mathfield/pointer-input.ts]: Is this going to prevent selection of read-only mathfields?

File: src/editor-mathfield/render.ts [hunk 4]
```diff
@@ -35,10 +35,15 @@ export function requestUpdate(
...
-      if (isValidMathfield(mathfield) && mathfield.dirty) {
+      if (
+        isValidMathfield(mathfield) &&
+        mathfield.dirty &&
+        !mathfield.options.readOnly
+      ) {
         mathfield._atomBoundsCache = new Map<string, Rect>();
         render(mathfield, options);
         mathfield._atomBoundsCache = undefined;
+        mathfield.attachNestedMathfield();
```
Comment (arnog) [inline src/editor-mathfield/render.ts]: Is this right? It does the attach only if the mathfield is *not* readonly?

File: src/public/mathfield-element.ts [hunk 5]
```diff
@@ -1036,16 +1026,26 @@ export class MathfieldElement extends HTMLElement implements Mathfield {
     // This.setAttribute('aria-multiline', 'false');
     if (!this.hasAttribute('tabindex')) this.setAttribute('tabindex', '0');
 
+    const slot =
```
Comment (arnog) [inline src/public/mathfield-element.ts:1030]: Why did you have the change the order?

File: src/editor-mathfield/pointer-input.ts [hunk 6]
```diff
@@ -19,7 +19,12 @@ export function onPointerDown(
 ): void {
   //Reset the atom bounds cache
   mathfield._atomBoundsCache = new Map<string, Rect>();
-
+  if (
+    (evt.target as HTMLElement).tagName === 'MATH-FIELD' &&
```
Comment (arnog) [inline src/editor-mathfield/pointer-input.ts]: the case of the tag is not guaranteed to be uppercase (it's lowercase in some browsers). So you need to do a case-insensitive comparison, i.e. `tagName.toLowerCase() === 'math-field'`

File: src/public/mathfield-element.ts (hunk 5 above)
Comment (arnog) [reply src/public/mathfield-element.ts:1030]: OK, I understand why now. I think it's OK, but please test with both Safari and Firefox.

File: src/editor-mathfield/mode-editor-math.ts (hunk 2 above)
Comment (caleb-flores) [reply src/editor-mathfield/mode-editor-math.ts]: I needed to scale down the size of the font because the mathfield was bigger than required.

File: src/editor-mathfield/pointer-input.ts (hunk 3 above)
Comment (caleb-flores) [reply src/editor-mathfield/pointer-input.ts]: the intention was to avoid triggering the re-rendering when you click over a nested mathfield, it didn't work because the main mathfield lose focus which is also causing re-rendering, I need to remove this.

File: src/editor-mathfield/render.ts (hunk 4 above)
Comment (caleb-flores) [reply src/editor-mathfield/render.ts]: you're right.

File: src/editor-mathfield/keyboard-input.ts (hunk 1 above)
Comment (caleb-flores) [reply src/editor-mathfield/keyboard-input.ts:366]: it was not intentional
